### PR TITLE
Shortcut refactoring

### LINKED ISSUE #161: Refactor shortcuts code
There is that file `static/shortcuts.js` that handles the shortcut and the shortcut cheatsheet logic. It is actually hard to understand, at least for me. I want there to be a refactoring. Myself, I tried to do it a couple of times, but always ended up getting lost and dropping it, like today.

So, I want the following:

* Make the code shorter. Currently, there are 370 lines.
* Make the code more declarative.
* Do not use that `<template>`, remove it. If you need the template, just construct it in-script. It makes more sense.

Extra goals:
* Make it extensible. Provide some global function (`window.rrhAddShortcut`, maybe) for customization. Currently, I have no idea how to add my own shortcut on my wiki. Maybe it is possible.
* Localize it. Currently, only the close button's `aria-label` is localized.

### PR DESCRIPTION
**Pull request by handlerug**
Resolves #161. This is the moment to comment my shit code

### CODE REVIEW

**Comment by handlerug** [comment]
Wow CodeQL you are actually useful! But let me disappoint you: this code is not used anywhere so shut up

**Comment by bouncepaw** [comment]
> Wow CodeQL you are actually useful! But let me disappoint you: this code is not used anywhere so shut up

Rude

**Comment by handlerug** [comment]
Please comment the overall code structure and code readability. I've split some classes up, converted useless classes into functions, and made a public API for third-party shortcut additions. I've also added a simple l10n mechanism like the one used in your Go code. I'll continue working iteratively based on your commentary.

**Comment by handlerug** [comment]
Though I'm not sure what you meant by "more declarative". This API is still imperative, and I don't really know how to make it declarative. (A global array like rrh.shortcuts you push your shortcuts to, maybe? But then it'll need a bunch of Proxies to track changes and that'd basically be some VDOM implementation but for shortcuts lol)

**Comment by bouncepaw** [comment]
1. When opening the shortcut dialog, the selection border (pink) has a weird shape.

   Снимок экрана 2022-08-21 в 17 04 30
2. The close button does not work. Escape button works.
3. Liking the i10n approach.
4. Liking the shortcut graph.
5. I don't like `ShortcutGroup`'s `target`. It can be either a Function or a Node, I think it's better to limit it to just Function. Want to work on a Node? Make a function that makes a callback that you pass there. Well, you basically do the same with a recursive `bind` call. Maybe that's better.
6. Line 266 and below don't make much sense to me. I expect something more like that:
   ```js
   rrh.addShortcutGroup(new ShortCutGroup(/* name and whatnot */,
      new Binding(/* keys */, /*description */, /* callback */),
      new Binding(/* keys */, /*description */, /* callback */),
      new Binding(/* keys */, /*description */, /* callback */),
   ))
   ```
   Looks declarative to me. How does an end-user modify the list? With a separate function:
   ```js
   rrh.addBindingToGroup("group name", new Binding(/* ... */))
   ```
   Your approach is similar to having only the latter function.
7. > But then it'll need a bunch of Proxies to track changes and that'd basically be some VDOM implementation but for shortcuts lol)

   I don't see why.

**Comment by handlerug** [comment]
I agree with you except for the following points:

> 5\. I don't like `ShortcutGroup`'s `target`. It can be either a Function or a Node, I think it's better to limit it to just Function. Want to work on a Node? Make a function that makes a callback that you pass there. Well, you basically do the same with a recursive `bind` call. Maybe that's better.

It may get too wordy and I'm trying to avoid that. I think it's fine as it is, right now it's both explicit and succinct.

> 6\. Line 266 and below don't make much sense to me. I expect something more like that: ...

To be honest, the code snippet that follows your comment reeks of weird OOP patterns from something like Java. Classes are not the easiest thing to instantiate in JavaScript, and they're not The Idiom of JavaScript. Usually objects are used for this. Anyway, I get your idea, your set of methods might be a better solution now that I've consolidated all methods into a single `bind()` method.

P.S. By the way, do you like `rrh.l10nify`? It won't work with your suggestion; you'd need to sprinkle `rrh.l10n()` for every string in there. I'll wait for your response before proceeding.

**Comment by bouncepaw** [comment]
Re: 5. OK.

Re: P. S. OK.

**Comment by handlerug** [comment]
What do you mean by OK regarding the P.S.?

**Comment by bouncepaw** [comment]
It means I'm OK with sprinkling `rrh.l10n`

**Comment by handlerug** [comment]
Should addBindingToGroup fail if the group is not present? For example, if the shortcut for the editor is added on every page.

**Comment by bouncepaw** [comment]
It should

**Review by bouncepaw — APPROVED** [review]
LGTM 💛
